src/WCSimPMTObject.cc

```diff
@@ -1965,3 +2011,46 @@ G4double PMT3inch::GetDarkRateConversionFactor(){
   const G4double factor = 1.367;
   return factor;
 }
+
+G4int PMT3inch::GetNbOfQEDefined(){
+  const G4int factor = 20;
+  return factor;
+}
+
+////////////////////////////////////////////////////////////////////////////////////////////////
+// COMBINED PMTS CLASS
+// Goal is to instantiate purely virtual WCSimPMTObject Classes to contained an extended
+//  QE information of several PMTs collection
+// Useful for applying QE in stacking action when several PMTs collection are defined
+
+
+WCSimBasicPMTObject::WCSimBasicPMTObject(){
+}
+
+WCSimBasicPMTObject::WCSimBasicPMTObject(std::map<G4double,G4double> mQE){
+  mapQE=mQE;
+  std::map<G4double, G4double>::iterator itr;
+  for(itr = mQE.begin(); itr != mQE.end(); itr++) {
+    wavelength.push_back(itr->first);
+    QE.push_back(itr->second);
+  }
+}
+
+WCSimBasicPMTObject::WCSimBasicPMTObject(std::vector<G4double> wl,std::vector<G4double> qe,G4double max){
+  wavelength=wl;
+  QE=qe;
+  maxQE=max;
+}
+
+void WCSimBasicPMTObject::DefineQEHist(std::map<G4double,G4double> mapQE){
+  gQE = new TGraph();
```

Comment:
```suggestion
  gQE = new TGraph(mapQE.size());
```
should be more efficient.
Please also check if the TGraph exists